public/app/features/dashboard/components/PanelEditor/OptionsGroupWithPersistence.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { FC, useCallback } from 'react';
+import { useLocalStorage } from 'react-use';
+import { PANEL_EDITOR_UI_STATE_STORAGE_KEY } from './state/reducers';
+import { OptionsGroup } from './OptionsGroup';
+import { OptionsGroupProps } from './types';
+
+interface Props extends OptionsGroupProps {
+  id: string;
+}
+
+export const OptionsGroupWithPersistence: FC<Props> = props => {
+  const [value, setValue] = useLocalStorage(getOptionGroupStorageKey(props.id), {
+    defaultToClosed: props.defaultToClosed,
+  });
+  const onToggle = useCallback(
+    (isExpanded: boolean) => {
+      setValue({ defaultToClosed: !isExpanded });
+      if (props.onToggle) {
+        props.onToggle(isExpanded);
+      }
+    },
+    [setValue, props.onToggle]
+  );
```

Comment:
Would be great to have this as a reusable hook for other collapsible parts of the UI to reuse.

Reply:
Great suggestion and if I feel I have time for it, I'll do it!

Reply:
Yeah, that's nice to have if we squeeze transformations and query rows persistance in